model/fn-execution/src/main/proto/beam_fn_api.proto

```diff
@@ -266,6 +266,40 @@ message Metrics {
       map<string, int64> output_elements_remaining = 3;
     }
 
+    // User defined metrics
+    message User {
+
+      // A key for identifying a metric at the most granular level.
+      message MetricKey {
```

Comment:
<!--thread_id:cc_164923048_t; commit:ea8ec4f5bdb63253febaa6bcebf20e839c39749d; resolved:0-->
<!--section:context-quote-->
> **robertwb** wrote:
> We should probably be calling this a MetricName to avoid confusion with the MetricKey class in Java/Python that contains a step id.

<!--section:body-->
In that case, I've moved the runner translations to the MetricName class. Let me know what you think. I named it `metric_name`, because within it there's another `name` field.